### PR TITLE
Fix/41829/nav by

### PR DESCRIPTION
**Pull request by barryhughes**
Sometimes the admin toolbar obscures whatever we navigate to (links to specific sections of the importer setting page, or to the 'generate missing venue data' section etc). This change rectifies that problem.

:ticket: [#41829](https://central.tri.be/issues/41829)

### CODE REVIEW

File: src/resources/js/tribe-common.js [hunk 1]
```diff
@@ -85,6 +86,99 @@ tribe_auto_sysinfo.ajax = {
 
 	};
 
+	/**
+	 * Sets up listeners and callbacks to handle navigation to page #elements
+	 * gracefully and in a way that doesn't result in the admin toolbar obscuring
+	 * the target.
```
Comment (borkweb) [inline src/resources/js/tribe-common.js]: Add `@since`?

File: src/resources/js/tribe-common.js [hunk 2]
```diff
@@ -85,6 +86,99 @@ tribe_auto_sysinfo.ajax = {
...
+	 */
+	my.navigate_to_id = function() {
+		$( document ).ready( my.maybe_navigate_to_id_on_doc_ready );
+		$( '.tribe_events_page_tribe-common' ).click( my.maybe_navigate_to_id_after_click );
+	};
+
+	/**
+	 * When the document is ready, check and see if the current location included
+	 * a reference to a specific ID and trigger our offset/scroll position adjustment
+	 * code if so.
```
Comment (borkweb) [inline src/resources/js/tribe-common.js]: `@since`

File: src/resources/js/tribe-common.js [hunk 3]
```diff
@@ -85,6 +86,99 @@ tribe_auto_sysinfo.ajax = {
...
 	};
 
+	/**
+	 * Sets up listeners and callbacks to handle navigation to page #elements
+	 * gracefully and in a way that doesn't result in the admin toolbar obscuring
+	 * the target.
+	 */
+	my.navigate_to_id = function() {
+		$( document ).ready( my.maybe_navigate_to_id_on_doc_ready );
+		$( '.tribe_events_page_tribe-common' ).click( my.maybe_navigate_to_id_after_click );
```
Comment (borkweb) [inline src/resources/js/tribe-common.js]: Perhaps alter this to the following to bind to dynamically added elements (if they are added in the future):

```js
$( document ).on( 'click', '.tribe_events_page_tribe-common', my.maybe_navigate_to_id_after_click );
```

File: src/resources/js/tribe-common.js (hunk 2 above)
Comment (borkweb) [inline src/resources/js/tribe-common.js]: Rinse and repeat on the other docblocks.

File: src/resources/js/tribe-common.js (hunk 3 above)
Comment (barryhughes) [inline src/resources/js/tribe-common.js]: Oh good call, that would be way slicker

**Comment by barryhughes** [comment]
Added @since tags and other changes, thanks!

File: src/resources/js/tribe-common.js (hunk 1 above)
Comment (ggwicz) [inline src/resources/js/tribe-common.js]: confirming `@since` was added

File: src/resources/js/tribe-common.js (hunk 2 above)
Comment (ggwicz) [inline src/resources/js/tribe-common.js]: confirming `@since` was added

**Pull request merged**